src/tree_state.rs

```diff
@@ -201,27 +201,111 @@ where
     /// });
     /// ```
     ///
     /// For more examples take a look into the source code of [`key_up`](Self::key_up) or [`key_down`](Self::key_down).
     /// They are implemented with this method.
+    #[deprecated = "renamed to select_relative"]
     pub fn select_visible_relative<F>(&mut self, change_function: F) -> bool
     where
         F: FnOnce(Option<usize>) -> usize,
     {
-        let visible = &self.last_visible_identifiers;
+        let identifiers = &self.last_identifiers;
+        let current_identifier = &self.selected;
+        let current_index = identifiers
+            .iter()
+            .position(|identifier| identifier == current_identifier);
+        let new_index = change_function(current_index).min(self.last_biggest_index);
+        let new_identifier = identifiers.get(new_index).cloned().unwrap_or_default();
+        self.select(new_identifier)
+    }
+
+    /// Move the current selection with the direction/amount by the given function.
+    ///
+    /// Returns `true` when the selection changed.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use tui_tree_widget::TreeState;
+    /// # type Identifier = usize;
+    /// # let mut state = TreeState::<Identifier>::default();
+    /// // Move the selection one down
+    /// state.select_relative(|current| {
+    ///     // When nothing is currently selected, select index 0
+    ///     // Otherwise select current + 1 (without panicing)
+    ///     current.map_or(0, |current| current.saturating_add(1))
+    /// });
+    /// ```
+    ///
+    /// For more examples take a look into the source code of [`key_up`](Self::key_up) or [`key_down`](Self::key_down).
+    /// They are implemented with this method.
+    pub fn select_relative<F>(&mut self, change_function: F) -> bool
+    where
+        F: FnOnce(Option<usize>) -> usize,
+    {
+        let identifiers = &self.last_identifiers;
         let current_identifier = &self.selected;
-        let current_index = visible
+        let current_index = identifiers
             .iter()
             .position(|identifier| identifier == current_identifier);
         let new_index = change_function(current_index).min(self.last_biggest_index);
-        let new_identifier = visible.get(new_index).cloned().unwrap_or_default();
+        let new_identifier = identifiers.get(new_index).cloned().unwrap_or_default();
         self.select(new_identifier)
     }
 
-    /// Ensure the selected [`TreeItem`] is visible on next render
+    /// Select the node that was rendered for the given position on last render.
+    ///
+    /// Returns `true` when the selection changed.
+    #[must_use]
+    pub fn select_at(&mut self, position: Position) -> bool {
+        if let Some(identifier) = self.rendered_at(position) {
+            if identifier == self.selected {
+                false
+            } else {
+                self.select(identifier.to_vec())
+            }
+        } else {
+            false
+        }
+    }
+
```

Comment:
Remove `select_at`. Either `click_at` does exactly what is needed or `rendered_at` allows for more control over what is happening. When its useful it can still be added in the future.

```suggestion
```